packages/commerce-server/src/get-active-merchant-coupon.ts

```diff
@@ -6,9 +6,9 @@ export async function getActiveMerchantCoupon({
   siteCouponId,
   code,
 }: {
-  productId: string
-  siteCouponId: string
-  code: string
+  productId: string | undefined
+  siteCouponId: string | undefined
+  code: string | undefined
```

Comment:
The downstream uses of these values are each `string | undefined`, so I loosened this type signature so that I could use it more cleanly in the trpc query.

Reply:
making them optional is effectively the same result but less verbose with `?` so wondering if this was required/preferred?

Reply:
They are the same as far as I can tell and `code?: string`, etc. would have been better. Just typed it out as such probably right after looking at a type overlay that displayed as `string | undefined`.